src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc

```diff
@@ -0,0 +1,2125 @@
+
+    SubscribeForSignal();
+    command->Run();
+    utils::WaitForSdlExecute();
+    EXPECT_CALL(app_mngr_, SetUnregisterAllApplicationsReason(_)).Times(0);
+    EXPECT_CALL(app_mngr_, HeadUnitReset(_)).Times(0);
+#endif
+  }
+}
+
+TEST_F(HMICommandsNotificationsTest,
+       OnExitAllApplicationsNotificationReasonSuspend) {
+  MessageSharedPtr message = CreateMessage();
+  (*message)[am::strings::msg_params][am::hmi_request::reason] =
+      hmi_apis::Common_ApplicationsCloseReason::SUSPEND;
+
+  utils::SharedPtr<am::commands::Command> command =
+      CreateCommand<am::commands::OnExitAllApplicationsNotification>(message);
+
+  MessageSharedPtr ethalon_message =
+      CreateMessage(am::smart_objects::SmartType_Map);
+  (*ethalon_message)[am::strings::params][am::strings::function_id] =
+      hmi_apis::FunctionID::BasicCommunication_OnSDLPersistenceComplete;
+  (*ethalon_message)[am::strings::params][am::strings::message_type] =
+      am::MessageType::kNotification;
+  (*ethalon_message)[am::strings::params][am::strings::correlation_id] =
+      kCorrelationId_;
+  MessageSharedPtr temp_message = CreateMessage();
+
+  EXPECT_CALL(app_mngr_, GetNextHMICorrelationID())
+      .WillOnce(Return(kCorrelationId_));
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_))
+      .WillOnce(GetMessage(temp_message));
+  command->Run();
+  EXPECT_CALL(app_mngr_, GetNextHMICorrelationID()).Times(0);
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_)).Times(0);
+  EXPECT_EQ(
+      (*temp_message)[am::strings::params][am::strings::function_id].asInt(),
+      hmi_apis::FunctionID::BasicCommunication_OnSDLPersistenceComplete);
+  EXPECT_EQ(
+      (*temp_message)[am::strings::params][am::strings::message_type].asInt(),
+      am::MessageType::kNotification);
+  EXPECT_EQ(
+      (*temp_message)[am::strings::params][am::strings::correlation_id].asInt(),
+      kCorrelationId_);
+}
+
+TEST_F(HMICommandsNotificationsTest,
+       OnExitAllApplicationsNotificationReasonInvalidEnum) {
+  MessageSharedPtr message = CreateMessage();
+  (*message)[am::strings::msg_params][am::hmi_request::reason] =
+      hmi_apis::Common_ApplicationsCloseReason::INVALID_ENUM;
+
+  utils::SharedPtr<am::commands::Command> command =
+      CreateCommand<am::commands::OnExitAllApplicationsNotification>(message);
+
+  EXPECT_CALL(app_mngr_, SetUnregisterAllApplicationsReason(_)).Times(0);
+  EXPECT_CALL(app_mngr_, HeadUnitReset(_)).Times(0);
+  EXPECT_CALL(app_mngr_, GetNextHMICorrelationID()).Times(0);
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_)).Times(0);
+  command->Run();
+}
+
+TEST_F(HMICommandsNotificationsTest,
+       OnExitApplicationNotificationManageMobileCommand) {
+  MessageSharedPtr message = CreateMessage();
+  (*message)[am::strings::msg_params][am::strings::app_id] = kAppId_;
+  am::smart_objects::SmartObjectSPtr notification =
+      utils::MakeShared<am::smart_objects::SmartObject>();
+  (*notification)[am::strings::params][am::strings::function_id] =
```

Comment:
@LevchenkoS  @okozlovlux It is NOt redundant. You have to use static_cast to make conversion explicit.